pkg/cvo/internal/operatorstatus_test.go

```diff
@@ -0,0 +1,418 @@
+				}, {
+					Name: "operand-1", Version: "v0",
+				}},
+			},
+		},
+		exp: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+			Status: configv1.ClusterOperatorStatus{
+				Versions: []configv1.OperandVersion{{
+					Name: "operator", Version: "v1",
+				}, {
+					Name: "operand-1", Version: "v1",
+				}},
+			},
+		},
+		expErr: "cluster operator test-co is still updating: upgrading operand-1 from v0 to v1",
+	}, {
+		name: "cluster operator reporting desired operator and old versions for 2 operands",
+		actual: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+			Status: configv1.ClusterOperatorStatus{
+				Versions: []configv1.OperandVersion{{
+					Name: "operator", Version: "v1",
+				}, {
+					Name: "operand-1", Version: "v0",
+				}, {
+					Name: "operand-2", Version: "v0",
+				}},
+			},
+		},
+		exp: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+			Status: configv1.ClusterOperatorStatus{
+				Versions: []configv1.OperandVersion{{
+					Name: "operator", Version: "v1",
+				}, {
+					Name: "operand-1", Version: "v1",
+				}, {
+					Name: "operand-2", Version: "v1",
+				}},
+			},
+		},
+		expErr: "cluster operator test-co is still updating: upgrading operand-1 from v0 to v1, upgrading operand-2 from v0 to v1",
+	}, {
+		name: "cluster operator reporting desired operator and mix of old and desired versions for 2 operands",
+		actual: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+			Status: configv1.ClusterOperatorStatus{
+				Versions: []configv1.OperandVersion{{
+					Name: "operator", Version: "v1",
+				}, {
+					Name: "operand-1", Version: "v1",
+				}, {
+					Name: "operand-2", Version: "v0",
+				}},
+			},
+		},
+		exp: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+			Status: configv1.ClusterOperatorStatus{
+				Versions: []configv1.OperandVersion{{
+					Name: "operator", Version: "v1",
+				}, {
+					Name: "operand-1", Version: "v1",
+				}, {
+					Name: "operand-2", Version: "v1",
+				}},
+			},
+		},
+		expErr: "cluster operator test-co is still updating: upgrading operand-2 from v0 to v1",
+	}, {
+		name: "cluster operator reporting desired versions and no conditions",
+		actual: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+			Status: configv1.ClusterOperatorStatus{
+				Versions: []configv1.OperandVersion{{
+					Name: "operator", Version: "v1",
+				}, {
+					Name: "operand-1", Version: "v1",
+				}},
+			},
+		},
+		exp: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+			Status: configv1.ClusterOperatorStatus{
+				Versions: []configv1.OperandVersion{{
+					Name: "operator", Version: "v1",
+				}, {
+					Name: "operand-1", Version: "v1",
+				}},
+			},
+		},
+		expErr: "cluster operator test-co is not done; it is available=false, progressing=true, failing=true",
```

Comment:
I think this is too verbose - the previous message was more succinct (next user action can be to view the cluster operator).

Reply:
So to if the CO conditions do not match, we report just `cluster operator test-co is not done` .. ?

Reply:
The message reported is `fmt.Sprintf("Cluster operator %s has not yet reported success", actual.Name)` [here](https://github.com/openshift/cluster-version-operator/pull/105/files#diff-e4dff677d52fee4f39f3ba3c602f5659R182) and only the error is more detailed [here](https://github.com/openshift/cluster-version-operator/pull/105/files#diff-e4dff677d52fee4f39f3ba3c602f5659R178)

IMO this should be fine.

Reply:
Then we should check that in the test case too (table driven tests are great examples to reviewers)

Reply:
@smarterclayton done with https://github.com/openshift/cluster-version-operator/commit/d13c13903c1b92d11240fc628e0b6538b874f29b -> https://github.com/openshift/cluster-version-operator/commit/7098e9e7be4ee0a455c0aee6bc898284d5158bf4